Bottom: 0ff31ddc8f420fa015070d1cfaa2cd152825925d
Top:    b857c6917ad1e7f3e08d704b2880a2a697545152
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2013-01-03 19:57:47 +0000

find.c: Remove some "flag" variable usage in DRCFind()

Avoid confusion by passing *FLAG constants directly to some functions.
This removes several cases where we would set the variable, use it once,
then change it for the next usage, making the code more succinct.

I was not able to remove the flag variable entirely, as the while loop
calling DoIt() relies on it being changed inside the loop in order to
terminate. It should be possible to re-write the loop exit condition if
desired, but I'm not sure I completely understand the behaviour of the
existing code yet.

---

diff --git a/src/find.c b/src/find.c
index 2f4475f..05081b0 100644
--- a/src/find.c
+++ b/src/find.c
@@ -3338,33 +3338,28 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
   if (PCB->Shrink != 0)
     {
       Bloat = -PCB->Shrink;
-      flag = DRCFLAG | SELECTEDFLAG;
-      ListStart (What, ptr1, ptr2, ptr3, flag);
-      DoIt (flag, true, false);
+      ListStart (What, ptr1, ptr2, ptr3, DRCFLAG | SELECTEDFLAG);
+      DoIt (DRCFLAG | SELECTEDFLAG, true, false);
       /* ok now the shrunk net has the SELECTEDFLAG set */
       DumpList ();
-      flag = FOUNDFLAG;
-      ListStart (What, ptr1, ptr2, ptr3, flag);
+      ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
       Bloat = 0;
       drc = true;               /* abort the search if we find anything not already found */
-      if (DoIt (flag, true, false))
+      if (DoIt (FOUNDFLAG, true, false))
         {
           DumpList ();
           /* make the flag changes undoable */
-          flag = FOUNDFLAG | SELECTEDFLAG;
-          ClearFlagOnAllObjects (false, flag);
+          ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
           User = true;
           drc = false;
           Bloat = -PCB->Shrink;
-          flag = SELECTEDFLAG;
-          ListStart (What, ptr1, ptr2, ptr3, flag);
-          DoIt (flag, true, true);
+          ListStart (What, ptr1, ptr2, ptr3, SELECTEDFLAG);
+          DoIt (SELECTEDFLAG, true, true);
           DumpList ();
-          flag = FOUNDFLAG;
-          ListStart (What, ptr1, ptr2, ptr3, flag);
+          ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
           Bloat = 0;
           drc = true;
-          DoIt (flag, true, true);
+          DoIt (FOUNDFLAG, true, true);
           DumpList ();
           User = false;
           drc = false;
@@ -3396,12 +3391,10 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
     }
   /* now check the bloated condition */
   drc = false;
-  flag = FOUNDFLAG | SELECTEDFLAG;
-  ClearFlagOnAllObjects (false, flag);
+  ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
   Bloat = 0;
-  flag = SELECTEDFLAG;
-  ListStart (What, ptr1, ptr2, ptr3, flag);
-  DoIt (flag, true, false);
+  ListStart (What, ptr1, ptr2, ptr3, SELECTEDFLAG);
+  DoIt (SELECTEDFLAG, true, false);
   DumpList ();
   flag = FOUNDFLAG;
   ListStart (What, ptr1, ptr2, ptr3, flag);
@@ -3411,20 +3404,17 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
     {
       DumpList ();
       /* make the flag changes undoable */
-      flag = FOUNDFLAG | SELECTEDFLAG;
-      ClearFlagOnAllObjects (false, flag);
+      ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
       User = true;
       drc = false;
       Bloat = 0;
-      flag = SELECTEDFLAG;
-      ListStart (What, ptr1, ptr2, ptr3, flag);
-      DoIt (flag, true, true);
+      ListStart (What, ptr1, ptr2, ptr3, SELECTEDFLAG);
+      DoIt (SELECTEDFLAG, true, true);
       DumpList ();
-      flag = FOUNDFLAG;
-      ListStart (What, ptr1, ptr2, ptr3, flag);
+      ListStart (What, ptr1, ptr2, ptr3, FOUNDFLAG);
       Bloat = PCB->Bloat;
       drc = true;
-      DoIt (flag, true, true);
+      DoIt (FOUNDFLAG, true, true);
       DumpList ();
       drcerr_count++;
       LocateError (&x, &y);
@@ -3451,7 +3441,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
       IncrementUndoSerialNumber ();
       Undo (true);
       /* highlight the rest of the encroaching net so it's not reported again */
-      flag |= SELECTEDFLAG;
+      flag = FOUNDFLAG | SELECTEDFLAG;
       Bloat = 0;
       ListStart (thing_type, thing_ptr1, thing_ptr2, thing_ptr3, flag);
       DoIt (flag, true, true);
@@ -3462,8 +3452,7 @@ DRCFind (int What, void *ptr1, void *ptr2, void *ptr3)
     }
   drc = false;
   DumpList ();
-  flag = FOUNDFLAG | SELECTEDFLAG;
-  ClearFlagOnAllObjects (false, flag);
+  ClearFlagOnAllObjects (false, FOUNDFLAG | SELECTEDFLAG);
   return (false);
 }
